core/string/ustring.cpp

```diff
@@ -1653,13 +1653,13 @@ String String::num_scientific(double p_num) {
 
 #if defined(__GNUC__) || defined(_MSC_VER)
 
-#if (defined(__MINGW32__) || (defined(_MSC_VER) && _MSC_VER < 1900)) && defined(_TWO_DIGIT_EXPONENT) && !defined(_UCRT)
-	// MinGW and old MSC require _set_output_format() to conform to C99 output for printf
+#if defined(__MINGW32__) && defined(_TWO_DIGIT_EXPONENT) && !defined(_UCRT)
+	// MinGW requires _set_output_format() to conform to C99 output for printf
```

Comment:
Is this part still required with recent MinGW versions (2019 and later)?

Reply:
At lease MinGW-w64 9.0.0 on macOS is still using 3-digit exponent by default.
Output of the demo code from here - https://docs.microsoft.com/en-us/cpp/c-runtime-library/set-output-format?view=msvc-160
```
1.2110e-005 2.3056e-112
1.2110E-005 2.3056E-112
 1.211e-005  2.306e-112
 1.211E-005  2.306E-112
 1.2110e-05 2.3056e-112
 1.2110E-05 2.3056E-112
  1.211e-05  2.306e-112
  1.211E-05  2.306E-112
1.2110e-005 2.3056e-112
1.2110E-005 2.3056E-112
 1.211e-005  2.306e-112
 1.211E-005  2.306E-112
```